chemprop/utils.py

```diff
@@ -582,19 +602,28 @@ def save_smiles_splits(
     if not isinstance(smiles_columns, list):
         smiles_columns = preprocess_smiles_columns(path=data_path, smiles_columns=smiles_columns)
 
-    with open(data_path) as f:
```

Comment:
If possible, I would recommend keeping the `with` block pattern like I suggested in data/utils